### PR TITLE
warn if "make update" was not executed

### PR DESCRIPTION
**Pull request by stefantalpalaru**
and warn if Make's parallelism is not enabled

(needed for https://github.com/status-im/nimbus-eth2/issues/2140)

### CODE REVIEW

File: makefiles/targets.mk [hunk 1]
```diff
@@ -36,9 +38,24 @@ build:
...
+#- check if "make update" was executed
+warn-update:
+	if [[ -e $(UPDATE_TIMESTAMP) ]]; then \
+		if [[ $$(cat $(UPDATE_TIMESTAMP)) -ne $$($(GET_CURRENT_COMMIT_TIMESTAMP)) ]]; then \
+			echo -e "\n\"$$(basename "$(MAKE)") update\" was not executed for the current Git commit! The resulting build is unsupported.\n"; \
+		fi; \
+	fi
+
+#- check if we're enabling multiple jobs - https://stackoverflow.com/a/48865939
+warn-jobs:
```
Comment (arnetheduck) [inline makefiles/targets.mk:50]: this one probably only needs to be done once (by placing/depending on a .warned-jobs in `build/`)

File: makefiles/variables.mk [hunk 2]
```diff
@@ -66,33 +66,46 @@ GIT_STATUS := git status
...
 else
   PWD := pwd
+  EXE_SUFFIX :=
+  # available since Git 2.7.0
+  GIT_TIMESTAMP_ARG := --date=format-local:%s
 endif
+
+ifeq ($(shell uname), Darwin)
+  # md5sum - macOS is a special case
+  MD5SUM := md5 -r
```
Comment (arnetheduck) [inline makefiles/variables.mk:84]: do we use this somewhere?

File: makefiles/targets.mk (hunk 1 above)
Comment (stefantalpalaru) [reply makefiles/targets.mk:50]: Where do you see it multiple times?

File: makefiles/variables.mk (hunk 2 above)
Comment (stefantalpalaru) [reply makefiles/variables.mk:84]: https://github.com/status-im/nimbus-eth1/blob/de12fb3c9405d431e864db74a50034f113a83e73/Makefile#L103

File: makefiles/targets.mk (hunk 1 above)
Comment (stefantalpalaru) [reply makefiles/targets.mk:50]: Oh, you mean once per user. There's this other PR that removes any automagic build parallelism, making this warning rather important: https://github.com/status-im/nimbus-eth2/pull/2095

**Pull request merged**